packages/cli/src/lib/merge/README.md

```diff
@@ -0,0 +1,621 @@
+
+`ArrayExpression`'s meaningful contents are the elements in the array.
+`ObjectExpression`'s meaningful contents are its key/value pairs.
+`StringLiteral`'s meaningful content is its value.
+
+As of this writing, these are the only node types supported, but others may be added by updating `strategy.js`.
+
+### concatUnique
+Merges two AST nodes by appending the meaningful contents of the extension to the meaningful contents of the base, omitting duplicates. Meaningful content is defined on a per-node basis, and strives to be "what you expect":
+
+`ArrayExpression`'s meaningful contents are the elements in the array.
+`ObjectExpression`'s meaningful contents are its key/value pairs.
+
+As of this writing, these are the only node types supported, but others may be added by updating `strategy.js`.
+### interleave
+Merges two AST nodes like `concatUnique`, with the added permission to create and append AST nodes as needed. This is necessary when the straightforward concatenation of two AST nodes might produce invalid Javascript. The primary (as of this writing, only) use case is for `ImportDeclarations`, which have a few odd rules that make the following invalid:
+
+```js
+import *, { foo } from 'source'
+import *, defaultImport from 'source'
+import defaultImport, * from 'source'
+```
+
+Arguably, this strategy might just be thought of as the `concatUnique` implementation for `ImportDeclarations`, but the author believes the ability to create new AST nodes makes `interleave` conceptually distinct from `concatUnique`.
+
+As of this writing, only `ImportDeclarations` can be `interleave`d.
+### opaquely
+This is a strategy-modifier. It is a function which takes a strategy, and returns a strategy. In this particular case, `opaquely` simply marks the given strategy as "opaque", meaning that the strategy handles the merge for a particular AST node (say, a `FunctionDeclaration`) and the children of that type of node should not be recursively merged by other strategies.
+
+For example, given the following strategy:
+
+```js
+const strategy = {
+  ArrayExpression: concatUnique
+  FunctionDeclaration: opaquely((lhs, rhs) => lhs.body = [...lhs.body, ...rhs.body])
+}
+```
+
+The following demonstrates a simple merge:
+
+```js
+/* base.js */
+const x = [1, 2, 3]
+function y(a) {
+  const arr = ['a', 'b', 'c']
+  return arr.includes(a)
+}
+/* extension.js */
+const x = [3, 4, 5]
+function y(a) {
+  const arr = ['c', 'd', 'e']
+  return arr.includes(a)
+}
+/* merge.js */
+const x = [1, 2, 3, 4, 5]
+function y(a) {
+  const arr = ['a', 'b', 'c']
+  return y.includes(a)
+  const arr = ['c', 'd', 'e']
+  return y.includes(a)
+}
+```
+
+Notice that the array `arr` is not merged before the bodies of both versions of `y` are merged. `opaquely` tells the merge algorithm that the strategy provided (`(lhs, rhs) => lhs.body = [...lhs.body, ...rhs.body]`) is all that should be done to merge the contents of `FunctionDeclaration`s.
```

Comment:
You should get the Pulitzer Prize or something for this README. Thank you so much for writing it. It's incredibly useful and super well written!